src/app/components/forms/add/add-money-to-piggy/add-money-to-piggy.component.ts

```diff
@@ -0,0 +1,89 @@
+import { Component, EventEmitter, Input, Output } from '@angular/core';
+import { FormBuilder, FormGroup, Validators } from '@angular/forms';
+import { Account } from 'src/app/models/Account.model';
+import { PiggyBank } from 'src/app/models/PiggyBank.model';
+import { PaymentsService } from 'src/app/services/payments.service';
+import { PiggyBankService } from 'src/app/services/piggyBankService/piggy-bank.service';
+
+@Component({
+  selector: 'app-add-money-to-piggy',
+  templateUrl: './add-money-to-piggy.component.html',
+  styleUrls: ['./add-money-to-piggy.component.scss', '../../add/account-add/account-add.component.scss']
+})
+export class AddMoneyToPiggyComponent {
+  @Input() piggyInfo : PiggyBank;
+  @Output() closeForm = new EventEmitter<void>();
+  @Output() reInitialise = new EventEmitter<void>();
+
+  piggyForm!: FormGroup;
+
+  constructor(
+    private formBuilder: FormBuilder,
+    private piggyService: PiggyBankService,
+    private paymentService: PaymentsService
+  ) {}
+
+  ngOnInit(): void {
+    this.initializeForm();
+  }
+  close(){
+    this.closeForm.emit();
+  }
+
+  initializeForm(): void {
+    this.piggyForm = this.formBuilder.group({
+      goal: [{ value: this.piggyInfo.goal, disabled: true }],
+      goalAmount: [{ value: this.piggyInfo.goalAmount, disabled: true}],
+      saveAmount: ['', Validators.required]
+    });
+  }
+
+  onSubmit(): void {
+    this.closeForm.emit();
+    if (this.piggyForm.valid) {
```

Comment:
Please avoid unnecessary nesting with IF's.
It is better to check a negative scenario, than continue just with your logic:
```suggestion
    if (!this.piggyForm.valid) {
        return;
    }
```